### ISSUE TITLE
PDF: spacing of META inside of JAVASCRIPTINLINE

### ISSUE THREAD

**Issue by martin-henz** [source-academy/sicp#446]
META inside of JAVASCRIPTINLINE produces too much horizontal space after a blank:

<img width="823" alt="Screenshot 2021-02-07 at 8 21 07 PM" src="https://user-images.githubusercontent.com/30220768/107146295-273e6e80-6982-11eb-913f-9ba65ed526cb.png">

**Comment by martin-henz** [comment]
This is just a regular fixed-font blank. Nothing wrong, really!

**Comment by martin-henz** [comment]
@TobiasWrigstad I suggest we delete this issue. Maybe a feature, not a bug?

**Comment by TobiasWrigstad** [comment]
I think the simplest fix here is to merge the META tags so they include the space which will then be typeset with the italic font which is much smaller. Let's keep this one open yet and I'll get it done when more pressing issues are fixed.

**Comment by martin-henz** [comment]
I've tried that. It didn't change the size of the space.

**Comment by TobiasWrigstad** [comment]
What are you doing up??

**Comment by TobiasWrigstad** [comment]
I'll experiment and get back.

**Comment by martin-henz** [comment]
yea maybe the moon. probably too cloudy in stockholm to see it: It must have just risen.

**Comment by TobiasWrigstad** [comment]
Haha. Yeah. Definitely too cloudy here.

**Comment by TobiasWrigstad** [comment]
![image](https://user-images.githubusercontent.com/364888/116310143-eb10f200-a7a9-11eb-9425-54335834eb2a.png)

I've changed META tags to replace all spaces with thin spaces. This will not break anything as no existing META tags have spaces in them (verified using regex over the entire book just now). The first bullet illustrates the consequence of turning `(<META>x</META>, <META>y</META>)` into `(<META>x, y</META>)`.

**Comment by TobiasWrigstad** [comment]
Notably, the `,` renders as times. Do we consider that too ugly? Technically that part isn't META.

**Comment by TobiasWrigstad** [comment]
I am keeping this change as a git stash that I can apply or simply drop. Letting this rest for now.

**Comment by martin-henz** [comment]
I think your solution is good. Before you push to master, Check how the web version looks:
yarn web
yarn prepare
yarn try
then check http://localhost:8080/chapters/4.1.3.html

**Comment by martin-henz** [comment]
I can't reproduce your observation. I'm getting this

<img width="571" alt="Screenshot 2021-04-28 at 9 44 39 PM" src="https://user-images.githubusercontent.com/30220768/116414678-3c34ea80-a86b-11eb-9658-7456720f873c.png">

from this

<img width="659" alt="Screenshot 2021-04-28 at 9 45 31 PM" src="https://user-images.githubusercontent.com/30220768/116414701-448d2580-a86b-11eb-90fb-b7d9d8aa258a.png">

in both master and mitpress_7x10_rebased

**Comment by TobiasWrigstad** [comment]
I stashed it, I didn't apply it. So it is not in the branch on GitHub. I'll test the web version now and if that looks good I'll apply the stash and push.

**Comment by TobiasWrigstad** [comment]
![image](https://user-images.githubusercontent.com/364888/116463443-fbd56c80-a86a-11eb-85ca-59442de062a6.png)
Something's broken. I started by trying to establish a baseline from what was in the branch already and that does not render the symbols in META at all. I'll investigate now.

**Comment by TobiasWrigstad** [comment]
Hmm, I wonder. This "fixes" the problem (of not rendering the symbols at all):
```
diff --git a/javascript/parseXmlHtml.js b/javascript/parseXmlHtml.js
index 6d6916e5..54a808a8 100755
--- a/javascript/parseXmlHtml.js
+++ b/javascript/parseXmlHtml.js
@@ -342,12 +342,12 @@ const processTextFunctionsDefaultHtml = {

   JAVASCRIPTINLINE: (node, writeTo) => {
     if (ancestorHasTag(node, "NAME")) {
-      recursiveProcessPureText(node.firstChild, writeTo, {
+      recursiveProcessTextHtml(node.firstChild, writeTo, {
         removeNewline: "all"
       });
     } else {
       writeTo.push("<kbd>");
-      recursiveProcessPureText(node.firstChild, writeTo, {
+      recursiveProcessTextHtml(node.firstChild, writeTo, {
         removeNewline: "all"
       });
       writeTo.push("</kbd>");
```
Didn't we discuss this very bug in the past at some point? I'm getting a feeling of deja vu.

**Comment by martin-henz** [comment]
The current version in mitpress_7x10_rebased violates our new rule for the
tags:

      <JAVASCRIPT>
<UL>
 <LI>
   <JAVASCRIPTINLINE>lookup_symbol_value(<META>symbol</META>,
<META>env</META>)</JAVASCRIPTINLINE>
   <INDEX><USE>lookup_symbol_value</USE></INDEX>
 <BREAK/>

Here, we have a META within a JAVASCRIPTINLINE, which is no longer allowed.
Also: I thought we want to put the comma inside the META. So, we should
have:

      <JAVASCRIPT>
<UL>
 <LI>
   <JAVASCRIPTINLINE>lookup_symbol_value(</JAVASCRIPTINLINE><META>symbol,
env</META><JAVASCRIPTINLINE>)</JAVASCRIPTINLINE>
   <INDEX><USE>lookup_symbol_value</USE></INDEX>
 <BREAK/>

-martin
http://www.comp.nus.edu.sg/~henz

**Comment by martin-henz** [comment]
Did you rebase mitpress_7x10_rebased from master? This should be fixed in master.

**Comment by TobiasWrigstad** [comment]
Hmm. I can't be sure. However, I just spent a fair chunk of time rebasing after merging your Ch 5 PR. Will try again now.

**Comment by TobiasWrigstad** [comment]
Doesn't seem like this is fixed in master. 
```
	    <kbd>lookup_symbol_value(, )</kbd>
```

**Comment by martin-henz** [comment]
That chapter 5 PR had a change in 4.1.3:

<img width="662" alt="Screenshot 2021-04-29 at 6 18 22 AM" src="https://user-images.githubusercontent.com/30220768/116479761-c2285400-a8b2-11eb-8c1d-ef91e6f9f92f.png">

I think I see the problem.

**Comment by TobiasWrigstad** [comment]
OK. I am about to build a PDF and send to Julie now and then go to bed. I'll pick this up tomorrow!

**Comment by martin-henz** [comment]
You should have rebased the Ch 5 PR with the latest master, before rebasing mitpress_7x10_rebased with the Ch 5 PR.

**Comment by martin-henz** [comment]
Yes, sure. No rush for the web stuff of mitpress_7x10_rebased.

**Comment by martin-henz** [comment]
> You should have rebased the Ch 5 PR with the latest master, before rebasing mitpress_7x10_rebased with the Ch 5 PR.

As a result, the current web version at
https://source-academy.github.io/sicp/chapters/4.1.3.html
is broken.

**Comment by martin-henz** [comment]
No worries. I'll fix 4.1.3 in both branches, and check also 5.5.2 and 5.5.6 where there are similar issues.

**Comment by martin-henz** [comment]
I've handled this by splitting it up properly, and let LaTeX decide how big the space should be:

	    <JAVASCRIPTINLINE>assign_symbol_value(</JAVASCRIPTINLINE><META>symbol</META><JAVASCRIPTINLINE>,</JAVASCRIPTINLINE> <META>value</META><JAVASCRIPTINLINE>,</JAVASCRIPTINLINE> <META>env</META><JAVASCRIPTINLINE>)</JAVASCRIPTINLINE>
